source/Octopus.Tentacle.Client/Retries/RpcCallNoRetriesHandler.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Octopus.Tentacle.Client.Retries
+{
+    public class RpcCallNoRetriesHandler
+    {
+        public async Task ExecuteWithNoRetries(
+            Func<CancellationToken, Task> action,
+            bool abandonActionOnCancellation,
+            TimeSpan abandonAfter,
+            CancellationToken cancellationToken)
+        {
+            await ((Func<CancellationToken, Task>)(async ct =>
```

Comment:
Just wondering, what is the purpose of wrapping the entire contents of this method in a lambda, and calling it straight away?

I suppose it starts a new thread context at that point, but we would create a new thread context as soon as we hit an `await` in this method, right?

There is likely a subtlety I'm missing. 

But if not, it would be good if we could simplify this method to this:
```
public async Task ExecuteWithNoRetries(
            Func<CancellationToken, Task> action,
            bool abandonActionOnCancellation,
            TimeSpan abandonAfter,
            CancellationToken cancellationToken)
        {
            if (!abandonActionOnCancellation)
            {
                await action(cancellationToken).ConfigureAwait(false);
            }

            using var abandonCancellationTokenSource = new CancellationTokenSource();
            using (cancellationToken.Register(() =>
                   {
                       // Give the actionTask some time to cancel on it's own.
                       // If it doesn't assume it does not co-operate with cancellationTokens and walk away.
                       abandonCancellationTokenSource.TryCancelAfter(abandonAfter);
                   }))
            {
                var abandonTask = abandonCancellationTokenSource.Token.AsTask();

                try
                {
                    var actionTask = action(cancellationToken);
                    var completedTask = await Task.WhenAny(actionTask, abandonTask).ConfigureAwait(false);
                    if (actionTask != completedTask)
                    {
                        actionTask.IgnoreUnobservedExceptions();
                    }

                    await completedTask.ConfigureAwait(false);
                }
                catch (Exception e) when (e is OperationCanceledException)
                {
                    if (abandonCancellationTokenSource.IsCancellationRequested)
                    {
                        throw new OperationAbandonedException(e, abandonAfter);
                    }

                    throw;
                }
            }
        }
```